qiskit/circuit/quantumcircuit.py

```diff
@@ -451,6 +451,14 @@ def metadata(self, metadata: Optional[dict]):
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+    def __repr__(self):
+        return (
+            f"<QuantumCircuit '{self.name}' with"
+            f" {self.num_qubits} qubits, {self.num_clbits} clbits"
+            f", {len(self._data)} instructions, {len(self._calibrations)} calibrations"
+            f", and global_phase={self._global_phase}>"
```

Comment:
We can say `"and {self.global_phase} global phase."` as well, I think - I don't think we need to make that one look like code when we don't for the others.

Reply:
QuantumCircuit has many subclasses, so `type(self).__name__` might be better.